client/src/Components/Layouts/Customers/index.jsx

```diff
@@ -92,35 +92,90 @@ export default class Customers extends Component {
             filteredcustomersName: filterCustomerName
         })
     }
+
+
+    showModal = () => {
+        this.setState({ visible: true });
+    };
+    handleCancel = () => {
+        this.setState({ visible: false });
+    };
+
+    handleCreate = () => {
+        const form = this.formRef.props.form;
+        form.validateFields((err, values) => {
+            if (err) {
+                return;
+            }
+            let addCustomer = {
+                name: values.name,
+                email: values.email,
+                phone: parseInt(values.prefixPhone + values.phone),
+                status: values.status,
+                address: values.address
+            }
+            fetch('api/v1/addcustomer', {
+                method: 'POST',
+                headers: { 'content-type': 'application/json' },
+                body: JSON.stringify(addCustomer)
+            }).then(res => res.json())
+                .then(result => {
+                    if (result.result) {
+                        let newcustomer = [...this.state.customers];
+                        newcustomer.push(result.result[0])
+                        let newallData = [...this.state.customers];
+                        newallData.push(result.result[0])
+                        this.setState({
+                            customers: newcustomer,
+                            allData: newallData
+                        })
+                    }
+                })
+            form.resetFields();
+            this.setState({ visible: false });
+        });
+    };
+
+    saveFormRef = formRef => {
+        this.formRef = formRef;
+    };
     render() {
-        if(this.state.customers)
-       { return (
-            <div className="containercustomers">
-                <Sidebar />
-                <div className="conatinercustomers__customer">
-                    <Navbar />
-                    <Header title='إدارة المستخدمين' Icon={<Icon type="team" />} />
-                    <div className='addcustomer'>
-                        <Button name='إضافة مستخدم' icon={<Icon type="user" />} />
-                        <div className="filtercontainer">
-                            <div classNam="filtercontainer__orderdate">
-                                <RangePicker
-                                    showTime={{ format: 'HH:mm' }}
-                                    format="YYYY-MM-DD HH:mm"
-                                    placeholder={['من', 'الى']}
-                                    onChange={e => this.filterfunction(e, '', 'date')}
-                                   className="containercustomers__customer-rangpicker"
-                                />
-                                <span className='filtercontainer__orderdate-date'>فلترة حسب الوقت</span>
+        if (this.state.customers) {
```

Comment:
if there is no data in the state this message will appear
```
Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
```

to solve it add else statment and create a loading page